src/Misc/fileTools.ts

```diff
@@ -493,6 +510,50 @@ export class FileTools {
     public static IsFileURL(): boolean {
         return typeof location !== "undefined" && location.protocol === "file:";
     }
+
+    /**
+     * Test if the given uri is a base64 string
+     * @param uri The uri to test
+     * @return True if the uri is a base64 string or false otherwise
+     */
+    public static IsBase64DataUrl(uri: string): boolean {
+        // Check that the length of the string is at least as long as the minimum base64 data url length
+        if (uri.length >= "data:;base64,".length)
+        {
+            if (uri.substr(0, 5) === "data:") {
+                let commaIndex = uri.indexOf(",");
+                return commaIndex > 7 && uri.substr(commaIndex - 7, 7) === ";base64";
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Decode the given base64 uri.
+     * @param uri The uri to decode
+     * @return The decoded base64 data.
+     */
+    public static DecodeBase64UrlAsBinary(uri: string): ArrayBuffer {
+        const decodedString = atob(uri.split(",")[1]);
+        const bufferLength = decodedString.length;
+        const bufferView = new Uint8Array(new ArrayBuffer(bufferLength));
+
+        for (let i = 0; i < bufferLength; i++) {
+            bufferView[i] = decodedString.charCodeAt(i);
+        }
+
+        return bufferView.buffer;
+    }
+
+    /**
+     * Decode the given base64 uri into a UTF-8 encoded string.
+     * @param uri The uri to decode
+     * @return The decoded base64 data.
+     */
+    public static DecodeBase64UrlAsText(uri: string): string {
+        return atob(uri.split(",")[1]);
+    }

```

Comment:
fileTools? it should be stringTools. we already have other base64 things in stringTools, but there is a problem with moving it to stringTools for treeshaking.

Reply:
I didn't realize StringTools had logic for decoding and encoding Base64 strings. I'm not sure what the advantage of that is versus `atob` and `btoa`. I could see using StringTools.Decode inside of these functions, but it feels kind of odd to add logic in StringTools for understanding the URL structure. I am fine with moving the implementation wherever you think is appropriate though.

Reply:
atob and btoa only deals with strings. It takes and returns strings. They don't work for binary data.

Reply:
> I could see using StringTools.Decode inside of these functions, but it feels kind of odd to add logic in StringTools for understanding the URL structure.

Yeah, that would be weird. I think the best would be if the fileTools function parsed the url and the stringTools decode. Similar for encode.

Reply:
Moved the string manipulation of Base64 data into StringTools now with the Data URL manipulation still being in FileTools.